cmd/execution_builder.go

```diff
@@ -902,10 +904,29 @@ func (exeNode *ExecutionNode) LoadIngestionEngine(
 // create scripts engine for handling script execution
 func (exeNode *ExecutionNode) LoadScriptsEngine(node *NodeConfig) (module.ReadyDoneAware, error) {
 	// for RPC to load it
+	vm := fvm.NewVirtualMachine()
+
+	options := computation.DefaultFVMOptions(node.RootChainID, false, false)
+	vmCtx := fvm.NewContext(options...)
+
+	derivedChainData, err := derived.NewDerivedChainData(derived.DefaultDerivedDataCacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	queryExecutor := query.NewQueryExecutor(
+		exeNode.exeConf.computationConfig.QueryConfig,
+		node.Logger,
+		metrics.NewExecutionCollector(node.Tracer),
+		vm,
+		vmCtx,
+		derivedChainData,
+	)
+
 	exeNode.scriptsEng = scripts.New(
 		node.Logger,
 		node.State,
-		exeNode.computationManager,
```

Comment:
I just noticed that it might be better to just use `exeNode.computationManager.queryExecutor` here. The reason for that is that `derivedChainData` (a.k.a.: the programs cache) is filled by transaction execution and used by script execution, but that only happens if the query executor uses the same `derivedChainData` as the computation manager. 

creating a new `derivedChainData` here basically means the scripts will always have an empty cache. This might be ok for archive nodes (which can be scaled by adding more archive nodes) but is not ok for execution nodes.

Reply:
I see. Thank you for pointing that out, makes sense.